### PR TITLE
Stop click propagation when input has values to avoid unchecking accidentally

### PR DESCRIPTION
**Pull request by wender**
#### What problem is this solving?

When a user enters customization and clicks it back to check or make any changes, it unchecks the attachment selection. 

#### How to test it?

[Problem](https://www.awesomescreenshot.com/video/11951834?key=d67e2f1fe93dfd4578bcb0c600b12e89)

[Working Fix](https://wender--brassbell.myvtex.com/7-inch-brass-engravable-wall-bell---antiqued/p)

### CODE REVIEW

File: react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx [hunk 1]
```diff
@@ -90,13 +90,28 @@ const ModalView: FC<{ closeAction: () => void }> = ({
...
-  modalProps?: unknown
+  modalProps?: any
 }
 
 interface ButtonProps {
   collapse?: 'left' | 'right' | 'none'
 }
 
+const hasValues = (itemChildren: any) => {
+  const [name] = Object.getOwnPropertyNames(itemChildren)
```
Comment (icazevedo) [inline react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx:101]: Why do you only check for the first item?

File: react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx [hunk 2]
```diff
@@ -90,13 +90,28 @@ const ModalView: FC<{ closeAction: () => void }> = ({
...
+  const inputs = itemChildren[name].inputValues
+
+  if (!itemChildren[name]?.valuesOfInputValues) {
```
Comment (icazevedo) [inline react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx:104]: Here you conditionally access the property (?.) and on line 102 you don't. Is there a reason for it?

File: react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx [hunk 3]
```diff
@@ -90,13 +90,28 @@ const ModalView: FC<{ closeAction: () => void }> = ({
...
+    return false
+  }
+
+  return (
+    inputs.findIndex((input: { label: string }) => {
+      return itemChildren[name].valuesOfInputValues[input.label] !== ''
```
Comment (icazevedo) [inline react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx:110]: How does this behave on numerical inputs?

**Review by icazevedo — COMMENTED** [review]
.

File: react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx (hunk 2 above)
Comment (wender) [reply react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx:104]: Good point, changing it.

File: react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx (hunk 3 above)
Comment (wender) [reply react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx:110]: Will work the same way

File: react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx (hunk 1 above)
Comment (wender) [reply react/components/ProductAssemblyOptions/ProductAssemblyOptionItemCustomize.tsx:101]: Because it's always the name

**Comment by icazevedo** [comment]
I still get some weird behaviors sometimes. If I click "Add Text", add nothing to the lines, click again and then add something to the lines, the checkbox is still unchecked.

If I click "Add Text", fill any line with any text, Click "Add Text" again, remove everything, then click "Add Text" again and enter any content, the checkbox is unchecked.

I don't think the logic is correct just yet. I think a more consistent way of doing this would be: If the checkbox is checked, never uncheck it by clicking the "Add Text" button.

Also, I didn't understand your reply saying "The first is always the name". Which name? In this store only or in any store?

**Comment by danzanzini** [comment]
Hello @wender, is this still needed?

**Pull request closed without merging**